### DISCUSSION TITLE
file menu new link variable

### DISCUSSION THREAD

**Discussion by imed-ghomari** [Taitava/obsidian-shellcommands#221] [Ideas]
Hi, I'm currently using the file menu event to change yaml of some files, and it's been working very nicely.

However, now, I'm using a new type of link (provided by dataview plugin) which includes a heading section like this `note > heading`.

The variables {{event_file_name}} and {{event_title}} return only the note title. So I was thinking maybe add a new variable {{event_link}} which would return the full link name, including the heading section (if possible) ?

Thank you.

**Comment by Taitava** [comment]
Hi, it's a really good idea! 🙂 I never came to think about it myself, but I can definitely see it would be useful.

I tried to do some research on Obsidian's API, but unfortunately I could not find a way to implement this, at least yet. I opened up a [discussion about it on Obsidian's forums](https://forum.obsidian.md/t/file-menu-can-i-get-link-subpath/37372). Let's hope it will get some answer! I'd really want to do this kind of improvement for the file menu. 👍

**Reply by imed-ghomari** [reply]
Thank you for the research and the feature request.

**Reply by imed-ghomari** [reply]
Just want to make sure, we're talking about the same type of link. Because the one used in dataview provides a popup with the full name.

https://user-images.githubusercontent.com/38974541/168000090-58e30606-9a33-44cb-accb-93c9bf58a846.mov

**Reply by Taitava** [reply]
Ok, I'm not familiar with those links created by Dataview, but I'd quess this improvement should work there, too, judging from the fact that the link seemed to work almost like normal links in Obsidian, i.e. it's able to trigger this file menu event.
